packages/node-nesting-scope/src/NodeFinder/ScopeAwareNodeFinder.php

```diff
@@ -52,10 +52,13 @@ public function findParentType(Node $node, array $allowedTypes): ?Node
      */
     public function findParent(Node $node, callable $callable, array $allowedTypes): ?Node
     {
+        $this->isBreakingNodeFoundFirst = false;
         $parentNestingBreakTypes = array_diff(ControlStructure::BREAKING_SCOPE_NODE_TYPES, $allowedTypes);
 
-        $this->isBreakingNodeFoundFirst = false;
-        $foundNode = $this->betterNodeFinder->findFirstPrevious($node, function (Node $node) use (
+        if ($this->isBreakingNodeFoundFirst) {
+            return null;
+        }
```

Comment:
probably caused by it changes the behaviour of this class, that `$this->isBreakingNodeFoundFirst` called first, moved up to before return of Method Call.